org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/QueueService.java

```diff
@@ -81,24 +83,42 @@
 	
 	/**
 	 * No argument constructor for OSGi
+	 * @throws URISyntaxException 
 	 */
 	public QueueService() {
-		
+		this(getQueueRootFromProperty(), CommandConstants.getScanningBrokerUri());
+	}
+	
+	private static final String getQueueRootFromProperty() {
+		String root = System.getProperty("org.eclipse.scanning.event.queues.queue.root");
+		if (root==null) root = System.getProperty("GDA/gda.event.queues.queue.root");
+		return root;
 	}
 	
+	public QueueService(String queueRoot, URI uri) {
+		this.queueRoot = queueRoot;
+		this.uri = uri;
+	}
 	/**
 	 * Constructor for tests
 	 */
-	public QueueService(String queueRoot, URI uri) {
-		QueueService.queueRoot = queueRoot;
-		QueueService.uri = uri;
+	public QueueService(String queueRoot, String uri) {
```

Comment:
As this is not for testing (from what I can tell) could you move the comment above this constructor.